pkg/shed/index.go

```diff
@@ -259,7 +271,11 @@ func (f Index) PutInBatch(batch *leveldb.Batch, i Item) (err error) {
 		f.logger.Debugf("keyfields encoding error in PutInBatch. Error: %s", err.Error())
 		return err
 	}
-	batch.Put(key, value)
+	err = batch.Set(key, value)
+	if err != nil {
+		f.logger.Debugf("could not set values in batch. Error : %s", err.Error())
```

Comment:
Why the error is not passed to the caller?

This may result in committing an incomplete batch.

Reply:
Do not log error and return it.